File: Deficient Numbers/DeficientNumbers.py
import math
def divisorSum(n):
        sum = 0
        j = 1
        while(j <= math.sqrt(n)):
                if(n % j == 0):
                        if(n / j == j):
                                sum = sum + j
                        else:
                                sum = sum + j
                                sum = sum + (n/j)
                j += 1
        return sum
        

def isDeficient(n):
        if(divisorSum(n) < (2 * n)):
                ans.append(n)
                return True
        return False

n = int(input("Enter the nth value to find : "))
count = 0
ans = []
i = 0
while(n >= count):
        if(isDeficient(i)):
                count += 1
        i += 1
print(ans)
        
        

File: Deficient Numbers/Deficient Numbers.java
import java.util.*;
public class Main 
{
    static int divsum(int n) 
    {
        int sum = 0;
        for (int i = 1; i <= (Math.sqrt(n)); i++) {
            if (n % i == 0) {
 
                if (n / i == i) {
                    sum = sum + i;
                } else {
                    sum = sum + i;
                    sum = sum + (n / i);
                }
            }
        }
        return sum;
    }
 
    static boolean isDeficientNumber(int n) 
    {
        return (divsum(n) < (2 * n));
    }
 
    public static void main(String args[]) {
        int n ;
        Scanner sc = new Scanner (System.in);
        n = sc.nextInt();
        for (int i =0 ; i < n ; i ++){
            if (isDeficientNumber(i))
                System.out.print(i+", ");
        }
    }
}
